File: Server/src/Dtos/AuthDto.cs
namespace Posts_Website.Dtos
{
    public record class LoginDto
    (
        string Email,
        string Password
    );

    public record class RegisterDto
    (
        string UserName,
        string Email,
        string Password
    );

    public record class AuthDto
    (
        bool IsSuccessful,
        string? JwtToken,
        string? Message,
        UserDto? User
    );
}

File: Client/src/typescript/dtos/AuthDto.ts
import { UserDto } from "./UserDto"

export interface AuthDto {
    isSuccessful: boolean,
    jwtToken: string | null,
    message: string | null,
    user: UserDto | null
}

export interface LoginDto {
    email: string,
    password: string
}

export interface RegisterDto {
    userName: string,
    email: string,
    password: string
}